os/binfmt/libelf/libelf_bind.c

```diff
@@ -378,11 +412,16 @@ int elf_bind(FAR struct elf_loadinfo_s *loadinfo, FAR const struct symtab_s *exp
 	}
 
 	/* Read the symbol table into memory */
-	elf_readsymtab(loadinfo);
+	if (elf_readsymtab(loadinfo) == -ENOMEM) {
+		elf_enable_caching(loadinfo);
+	}
 
-#ifdef CONFIG_SUPPORT_COMMON_BINARY
-	elf_readstrtab(loadinfo);
 
+	if (elf_readstrtab(loadinfo) == -ENOMEM) {
+		elf_enable_caching(loadinfo);
```

Comment:
cache is already initialized from line 416.
Is it okay to initialize the cache again?